cmake/winml_unittests.cmake

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set(WINML_TEST_SRC_DIR ${REPO_ROOT}/winml/test)
+set(WINML_TEST_INC_DIR
+  ${REPO_ROOT}/winml/test/common
+  ${REPO_ROOT}/winml/lib/Api.Image/inc
+  ${REPO_ROOT}/winml/lib/Common/inc
+  ${REPO_ROOT}/onnxruntime
+  ${REPO_ROOT}/cmake/external/protobuf/src
+  ${CMAKE_CURRENT_BINARY_DIR}
+  ${CMAKE_CURRENT_BINARY_DIR}/winml_api
+  ${CMAKE_CURRENT_BINARY_DIR}/winml_api/comp_generated
+  ${CMAKE_CURRENT_BINARY_DIR}/winml/sdk/cppwinrt/include)
+
+function(AddWinMLTest)
+  cmake_parse_arguments(_UT "DYN" "TARGET" "LIBS;SOURCES;DEPENDS" ${ARGN})
+  if(_UT_LIBS)
+    list(REMOVE_DUPLICATES _UT_LIBS)
+  endif()
+  list(REMOVE_DUPLICATES _UT_SOURCES)
+
+  if (_UT_DEPENDS)
+    list(REMOVE_DUPLICATES _UT_DEPENDS)
+  endif(_UT_DEPENDS)
+
+  add_executable(${_UT_TARGET} ${_UT_SOURCES})
+  source_group(TREE ${WINML_TEST_SRC_DIR} FILES ${_UT_SOURCES})
+
+  set_target_properties(${_UT_TARGET} PROPERTIES
+    FOLDER "WinMLTest"
+    CXX_STANDARD 17
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+  )
+
+  if (_UT_DEPENDS)
+    add_dependencies(${_UT_TARGET} ${_UT_DEPENDS})
+  endif(_UT_DEPENDS)
+  target_link_libraries(${_UT_TARGET} PRIVATE ${_UT_LIBS} gtest_main windowsapp winml_lib_image onnx ${onnxruntime_EXTERNAL_LIBRARIES} libprotobuf) # TODO review if all of these are needed
```

Comment:
># TODO review if all of these are needed [](start = 148, length = 40)

can you do this before completing the PR?

Reply:
Yes, I'll look at all TODOs and FIXMEs